### PR TITLE
Single package - DO NOT MERGE

### PR DESCRIPTION
**Pull request by jkarni**
Outline of how to have a single package, with no flags. Just declares a new datatype and a `FromHttpApiData` instance for it.

@soenkehahn what do you think?

### CODE REVIEW

File: src/Servant/MatrixParam/Internal.hs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+-- >>> type MyApi = WithMatrixParams "book" '[MatrixParam "author" String] :> Get '[JSON] [String]
+type WithMatrixParams path params = Capture "matrixParamCapture" (MatrixParams path params)
+
+data MatrixParams (path :: Symbol) (paramSpecs :: [*]) where
+  MatrixEmpty :: MatrixParams path '[]
+  (:.:) :: Maybe a -> MatrixParams path as -> MatrixParams path (MatrixParam seg a ': as)
+    deriving (Typeable)
+infixr 3 :.:
+
+class Unmatrix ls a | ls -> a where
```
Comment (soenkehahn) [inline src/Servant/MatrixParam/Internal.hs]: This seems to be convenience functionality and should be clearly separated from the essential concepts, I think.

**Comment by soenkehahn** [comment]
I personally prefer the previous version. This PR automatically gives us lots of interpretations via `Capture`,  but I'm not sure that these interpretations will really be what we want. For example `servant-docs` and `servant-swagger` will mention captures with the name `matrixParamCapture` and no real information about how to specify them. Also the behavior of `servant-client` will change. Before, every matrix parameter resulted in a separate haskell parameter. Now we'll have one parameter of type `MatrixParams` (IIUC). Same goes for `servant-server`.

That being said, I'm not planning to use matrix parameters anytime soon, so feel free to just disregard my opinion.

**Comment by jkarni** [comment]
>  For example servant-docs and servant-swagger will mention captures with the name matrixParamCapture and no real information about how to specify them. 

`servant-docs` will require a `ToCapture`, where information about how matrix params works can be provided, which seems like the right thing to me. `servant-swagger`, likewise.

Yes, the behavior of servant-client and servant-server will change. That's something I'm indeed not very satisfied by, since the old behavior seems much nicer to me.

**Comment by jkarni** [comment]
Closing in favor of #2

**Pull request closed without merging**